todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/tasks/TaskItemViewModel.java

```diff
@@ -17,23 +17,29 @@
 package com.example.android.architecture.blueprints.todoapp.tasks;
 
 import android.content.Context;
+import android.support.annotation.Nullable;
 
 import com.example.android.architecture.blueprints.todoapp.TaskViewModel;
 import com.example.android.architecture.blueprints.todoapp.data.source.TasksRepository;
 
+import java.lang.ref.WeakReference;
+
 
 /**
  * Listens to user actions from the list item in ({@link TasksFragment}) and redirects them to the
  * Fragment's actions listener.
  */
 public class TaskItemViewModel extends TaskViewModel {
 
-    private final TaskItemNavigator mTaskItemNavigator;
+    @Nullable
+    private WeakReference<TaskItemNavigator> mNavigator;
```

Comment:
Because we're using a ListView, there is no way to clear the references for each item. Or at least there's no straightforward way.